### PR TITLE
papilo: remove accidental dependency

### PR DESCRIPTION
**Pull request by catap**
#### Description

Papilo may pick from system scip, soplex and HIGHS to be used as fronted which creates a depenency loop. Prevent that.

Also make clear dependency with OneTBB, GMP and OpenBLAS.

Also, enforce GMP instead libquadmath then the last one is available.

Thus, backport patch to support negative egcd.

###### Type(s)

- [x] bugfix

###### Tested on

macOS 12.6 21G115 x86_64
Xcode 14.1 14B47b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: math/papilo/Portfile [hunk 1]
```diff
@@ -5,8 +5,11 @@ PortGroup               boost 1.0
 PortGroup               cmake 1.1
 PortGroup               github 1.0
 PortGroup               compiler_blacklist_versions 1.0
+PortGroup               compilers 1.0
 
+# NOTE: PaPILO can be used as a header-based library, please bump all dependatd ports
```
Comment (reneeotten) [inline math/papilo/Portfile:10]: dependatd --> dependent

File: math/papilo/Portfile [hunk 2]
```diff
@@ -19,8 +22,28 @@ checksums               rmd160  086c74c8ce81f8b10c7073ea3b0090a7d1e77e0f \
...
 compiler.blacklist-append {clang < 1000}
 
-depends_lib-append      port:onetbb
+depends_lib-append      port:gmp \
+                        port:OpenBLAS \
+                        port:onetbb
+
+# PAPILO might be frontend for SCIP and/or SOPLEX
+# anyway, SCIP is also required PAPILO,
+# to broke the dependency loop and make build clean
```
Comment (reneeotten) [inline math/papilo/Portfile:38]: broke --> break

**Review by reneeotten — COMMENTED** [review]
looks good to me, two typos in the comments that you could correct; other than that, happy to merge

**Comment by catap** [comment]
This port is bump revision of `scip`. The last time it can't be build on arm64 because `ipopt` was broken. I've fixed it at https://github.com/macports/macports-ports/pull/16865

Anyway, I'd like to keep it as draft until `ipopt` is passed builds on buildbots.

**Comment by reneeotten** [comment]
@catap what's the status here; anything else needed for this to merged?

File: math/papilo/Portfile [hunk 3]
```diff
@@ -15,12 +18,30 @@ maintainers             {@catap korins.ky:kirill} openmaintainer
...
+                        size    1083482
 
+compilers.setup         require_fortran
 compiler.cxx_standard   2014
 
 compiler.blacklist-append {clang < 1000}
 
-depends_lib-append      port:onetbb
+depends_lib-append      port:gmp \
+                        port:OpenBLAS \
```
Comment (reneeotten) [inline math/papilo/Portfile:31]: I *think* typically the openblas dependency is specified such that the -devel port could also satisfy the dependency. Is that possible here as well (you can grep the ports to see how it's typically done).

Comment (catap) [reply math/papilo/Portfile:31]: Thanks. I’ve missed that a few months ago

**Comment by catap** [comment]
@reneeotten here it is

**Pull request merged**